SYS_ATL/API.py

```diff
@@ -519,10 +519,13 @@ def expand_dim(self, stmt_pat, alloc_dim_pat, indexing_pat):
         loopir = self._loopir_proc
         for i in range(0, stmts_len):
             s = self._find_stmt(stmt_pat, body=loopir.body, default_match_no=None)[i]
-            alloc_dim = parse_fragment(loopir, alloc_dim_pat, s)
-            indexing  = parse_fragment(loopir, indexing_pat, s)
+            alloc_dim = parse_fragment(loopir, alloc_dim_pat, None)
+            indexing  = parse_fragment(loopir, indexing_pat, None)
             loopir = Schedules.DoExpandDim(loopir, s, alloc_dim, indexing).result()
 
+        # Running checkeffect here is necessary for bounds checking
+        CheckEffects(loopir)
+
```

Comment:
Is this the line that's causing the Gemmini failures?

Reply:
It appears to be.

Reply:
Yeah, we need this line for the correctness of `expand_dim`